packages/grafana-runtime/src/services/dataSourceSrv.ts

```diff
@@ -75,6 +75,9 @@ export interface GetDataSourceListFilters {
 
   /** Only returns datasources matching the specified types (ie. Loki, Prometheus) */
   type?: string | string[];
+
+  /** Set to false to avoid returning '-- Grafana --' datasource */
+  grafana?: boolean;
```

Comment:
this parameter name is too generic to understand what is doing here, could you use the existing `filters` attribute to achieve this same?

Reply:
The part that adds the `-- Grafana --` datasource happens after the `filter` function so I'm not sure we could do it backwards-compatibly, unless I'm missing something (I don't see a `filters` attribute specifically :thinking:).

I could maybe rename it to `includeBuiltinGrafanaDatasource?` or something?

Reply:
`includeBuiltinGrafanaDatasource` that sounds more like it! (grafana is a very generic word inside the grafana code :D) 

Is there any harm on modifying the code so it removes it as part of the filter?

From my point of view, we already have a filter functionality in this method and we are adding a second filtering functionality for a very specific purpose instead of improving the existing one. 

I'd even argue it is better to have a second filtering stage if it is impossible to modify the existing one than creating a second filter parameter.

Reply:
Yeah we could do - I mainly wanted to follow the pattern used by `-- Mixed --` and `-- Dashboard --` since those look quite similar to `-- Grafana --`, but I don't really mind!

I've pushed up another commit which modifies this to check that the filter function doesn't return `false` for the `-- Grafana --` datasource at the time that we add it, and added a test. LMK what you think!